tools/generate-helpers/pg-table-bindings/list.go

```diff
@@ -28,7 +28,7 @@ func init() {
 		&storage.ProcessIndicator{}:       resources.Indicator,
 		&storage.ReportConfiguration{}:    resources.VulnerabilityReports,
 		&storage.SimpleAccessScope{}:      resources.Role,
-		&storage.TokenMetadata{}:          resources.APIToken,
+		&storage.TokenMetadata{}:          resources.Integration,
```

Comment:
This would not allow backwards-compatibility between resources when using postgres.
I see a few options here:
- making the postgres migration depend on permission grouping
- introduce a permissionchecker for stores touching resources being grouped
- change the scope check logic to only rely on resource groups
- change the scope check logic to rely on either resource groups or single resources depending on the target resource type.

Reply:
In my opinion, investigating in creating a permission checker for stores with grouped resources will be beneficial, although I'm not sure about the effort for this solution.
What would be the most ideal solution in your opinion? Minding that potentially there will be more resource deprecations in the future, I think we could already establish the generic way how to handle this going forward.

Reply:
Seeing how the `ForResourceHelper.ScopeChecker` function has been adjusted, I'd be tempted to rely on the ForResourceHelper to enforce scope checks in the generated store code.